pkg/controller/job/job_controller.go

```diff
@@ -1149,7 +1179,7 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 //     replacement pod. the entire Job is terminating the finalizer can be
 //     removed unconditionally.
 func canRemoveFinalizer(logger klog.Logger, jobCtx *syncJobCtx, pod *v1.Pod, considerPodFailed bool) bool {
-	if jobCtx.job.DeletionTimestamp != nil || jobCtx.finishedCondition != nil || pod.Status.Phase == v1.PodSucceeded {
+	if jobCtx.job.DeletionTimestamp != nil || jobCtx.failureCondition != nil || pod.Status.Phase == v1.PodSucceeded {
```

Comment:
Hm, I think we want to OR with `successCondition`. We will actually proceed with removal of the finalizers only if the `SucceessCriteriaMet` condition is recorded in the job status. At that point we don't care about the pods.

Reply:
Oh, you're right.
This is my bad, thanks.

Reply:
Done.

Reply:
Is there some test, at least at unit level for this?

Reply:
We don't have a dedicated test for the `canRemoveFinalizer`.
So, Let's add cases to `TestTrackJobStatusAndRemoveFinalizers` since the `TrackJobStatusAndRemoveFinalizers` calls `canRemoveFinalizer`.

Reply:
Done.